wordpress-mobile/WordPress-FluxC-Android
# Dashboard Cards: Store Dashboard Cards

Parent: [WordPress-Android#15498](https://github.com/wordpress-mobile/WordPress-Android/issues/15498)

This PR:
- Updates the post card response (based on the newly agreed structure).
- Updates the post card model.
- Adds the cards database access object (dao).
- Adds cards dao to database and Increments version.
- Stores dashboard cards into the database.
- Retrieves dashboard cards from the database.

-----

To test:

- There is actually nothing to test in this PR, just review again the updated `CardsRestClient` and `CardsStore` classes and make sure that the accompanying tests are working as expected.
- When [WordPress-Android#15497](https://github.com/wordpress-mobile/WordPress-Android/issues/15497) will get deployed and then the `ReleaseStack_CardsTest` test suite gets implemented, then triggering this connected test suite will be another way of verifying that this solution works as expected.

-----

To manual test:

- If you really want to test the solution:
  - I can provide you with a patch that connects the dashboard cards store with its source. Let me know!
  - Or, you can wait for the accompanying `WordPress-Android` PR to be created, which will implement the connection between the dashboard cards store and its source.
- However, mind the fact that in both the above cases you would need to:
  - Either apply the [My Site Dashboard: Cards Endpoint](https://code.a8c.com/D69742) differential on your sandbox, proxy your device on it. Let me know if you need any help with that!
  - Or, you can wait for the [My Site Dashboard: Cards Endpoint](https://code.a8c.com/D69742) differential to be deployed. Then, you would be able to test the solution as usual.

-----

Merge instructions:
- [x] ~~Wait for the new API endpoint to be deployed (see [D69742](https://code.a8c.com/D69742)).~~
- [x] Wait for the accompanying [WordPress-Android#15637](https://github.com/wordpress-mobile/WordPress-Android/pull/15637) PR to be reviewed (and possibly approved).
- [x] Remove the `[DO NOT MERGE]` from the title.
- [x] Remove the `[PR] Not Ready For Merge` label.
